File: fit.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <TRandom3.h>
#include <TApplication.h>
#include <TH1D.h>
#include <TGraph.h>
#include <TF1.h>
#include <TMath.h>
#include <Math/Minimizer.h>
#include <Math/Factory.h>
#include <Math/Functor.h>

using namespace std;

int main(){
  TApplication app("app",NULL,NULL);

  TH1D *h = new TH1D("h","",40,0,40);
  //implementare fitfun
  TF1 *fitfun=new TF1("fitfun","",0,50);

  ifstream file("dati_highstat.dat");
  
  double mass;
  while (ifile >> mass){
    h->Fill(mass);
  }

  h->Draw();
  app.Run(true);



  //implementare fit binned con metodo Fit
  TCanvas *c1 = new TCanvas(); 
  h->Draw();



  app.Run(true);


  //implementare fit unbinned con TTree
  TTree *t = new TTree();


  TCanvas *c2 = new TCanvas(); 
  h->Draw();
  
  app.Run(true);

}

File: fit.py
rom   ROOT    import *
from   iminuit import Minuit
import numpy   as     np
from   math    import *
import sys


gStyle.SetOptStat(0)
h      = TH1D("h","",40,0,40)
#implementare fitfun
fitfun = TF1("fitfun","",0,50)

for line in open("dati_highstat.dat"):
    mass = float(line)
    h.Fill(mass)
h.Draw()
gApplication.Run(True)


# implementare fit binned con metodo Fit
c1 = TCanvas() 
h.Draw()



gApplication.Run(True)


# implementare fit unbinned con TTree
t = TTree()


c2 = TCanvas()
h.Draw()


gApplication.Run(True)
